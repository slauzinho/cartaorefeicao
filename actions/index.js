import { FETCH_CARDS_REQUEST, FETCH_CARDS_SUCCESS } from './types';

export const fetchCardsRequest = () => ({
    type: FETCH_CARDS_REQUEST
})

export const fetchCardSuccess = cards => ({
    type: FETCH_CARDS_SUCCESS,
    cards: cards.map(card => JSON.parse(card))
})