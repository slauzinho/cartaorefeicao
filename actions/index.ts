import {
  FETCH_CARDS_REQUEST,
  FETCH_CARDS_SUCCESS,
  CHANGE_ACTIVE_INDEX,
  FETCH_SITE_SUCCESS,
  FETCH_SITE_FAILURE,
  FETCH_SITE_REQUEST,
  CardActionTypes,
} from './types';
import { Details } from '../types';

export const fetchCardsRequest = () => ({
  type: FETCH_CARDS_REQUEST,
});

export const fetchCardsSuccess = (cards: string[]): CardActionTypes => ({
  type: FETCH_CARDS_SUCCESS,
  cards: cards.map(card => JSON.parse(card)),
});

export const changeActiveIndex = (index: number): CardActionTypes => ({
  type: CHANGE_ACTIVE_INDEX,
  index,
});

export const fetchSiteRequest = (
  cardNumber: string,
  password: string,
  email: string | undefined,
  tipo: string
): CardActionTypes => ({
  type: FETCH_SITE_REQUEST,
  payload: { cardNumber, password, email, tipo },
});

export const fetchSiteSuccess = (details: Details): CardActionTypes => ({
  type: FETCH_SITE_SUCCESS,
  payload: { ...details },
});

export const fetchSiteFailure = (error: string): CardActionTypes => ({
  type: FETCH_SITE_FAILURE,
  error,
});
