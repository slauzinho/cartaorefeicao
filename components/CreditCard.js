/* eslint-disable */

import React from 'react';
const { PropTypes, Component } = React;

import FlipCard from 'react-native-flip-card';
import { View,  Text, StyleSheet, Image } from 'react-native';
import frontCard from '../assets/images/front2.png';

export default class CreditCard extends Component {
  getValue(name) {
    return this[name]();
  }

  cvc() {
    if (!this.props.cvc) {
      return '•••';
    }
    return this.props.cvc.toString().length < 4
      ? this.props.cvc
      : this.props.cvc.toString().slice(0, 3);
  }

  number() {
    let string = '';

    if (!this.props.number) {
      string = '';
    } else {
      string = this.props.number.toString();
    }

    if (string.length > 16) string = string.slice(0, 16);
    while (string.length < 16) {
      string += '•';
    }
    for (let i = 1; i <= 3; i++) {
      let space_index = i * 4 + (i - 1);
      string = string.slice(0, space_index) + ' ' + string.slice(space_index);
    }

    return string;
  }

  render() {
    const cardStyle = [
      styles.container,
      { width: 300, height: 190, backgroundColor: 'white' }
    ];
    return (
      <View style={cardStyle}>
        <FlipCard
          style={[
            styles.container,
            { width: 300, height: 190, backgroundColor: this.props.bgColor }
          ]}
          friction={6}
          perspective={1000}
          flipHorizontal={true}
          flipVertical={false}
          clickable={true}
          flip={this.props.focused === true}
        >
          <View style={[styles.front, { width: 300, height: 190 }]}>
            <Image
              source={frontCard}
              style={[styles.bgImage, { width: 300, height: 190 }]}
            />
            <View style={styles.info}>
              <View style={styles.name}>
                <Text style={styles.textNumber}>{this.getValue('number')}</Text>
              </View>
            </View>
              <View style={styles.cardName}>
                <Text style={styles.textNumber}>{this.props.nome}</Text>
              </View>
          </View>
          <View style={[styles.front, { width: 300, height: 190 }]}>
            <Image
              source={require('../assets/images/Back.png')}
              style={[styles.bgImage, { width: 300, height: 190 }]}
            />
            {this.props.children}
            <View style={styles.cvc}>
              <Text style={styles.textCvc}>{this.getValue('cvc')}</Text>
            </View>
          </View>
        </FlipCard>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    borderWidth: 0,
    flex: null
  },
  bgImage: {
    position: 'absolute',
    left: 0,
    top: 0,
    bottom: 0,
    right: 0,
    borderRadius: 8
  },
  name: {
    flex: 2,
    paddingTop: 40,
    justifyContent: 'center'
  },
  textNumber: {
    color: '#fff',
    fontSize: 21,
    textAlign: 'center',
    marginBottom: 10,
    backgroundColor: 'transparent',
    textShadowColor: 'rgba(0, 0, 0, 0.75)',
    textShadowOffset: { width: 1, height: 1 },
    textShadowRadius: 1,
    fontFamily: 'monospace'
  },
  cvc: {
    width: 90,
    height: 30,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    right: 0,
    top: 115
  },
  textCvc: {
    color: '#000',
    fontSize: 18
  },
  info: {
    flex: 1
  },
  cardName: {
    position: 'absolute',
    left: 32,
    bottom: 20
  }
});

CreditCard.defaultProps = {
  number: null,
  cvc: null,
  name: '',
  expiry: '',
  focused: null,
  expiryBefore: 'month/year',
  expiryAfter: 'valid thru',
  shinyAfterBack: '',
  type: null,
  width: 300,
  height: 180,
  bgColor: '#F57A7A',
  clickable: true,
  nome: 'Cartão'
};
