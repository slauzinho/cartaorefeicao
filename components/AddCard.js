import React from 'react';
import axios from 'axios';
import HTMLParser from 'fast-html-parser';
import {
  StyleSheet,
  Text,
  View,
  Image,
  AsyncStorage,
  Picker,
  TouchableOpacity,
  ScrollView
} from 'react-native';
import { Input, Overlay, Button } from 'react-native-elements';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/FontAwesome';
import CreditCard from './CreditCard';
import { loginEdenred } from '../utilities/login';

const isNumeric = n => !isNaN(parseFloat(n)) && isFinite(n);

export default class AddCard extends React.Component {
  static navigationOptions = {
    title: 'Adicionar Cartão',
    headerStyle: {
      backgroundColor: '#FFF'
    },
    headerTintColor: '#000',
    headerTitleStyle: {
      fontWeight: 'bold'
    }
  };
  state = {
    cardNumber: '',
    cardPassword: '',
    tipo: 'santander',
    isFocused: false,
    isVisible: false,
    message: null,
    cardName: '',
    cardColor: '#F57A7A',
    isLoading: false
  };

  async onPressValidator() {
    this.setState({ isLoading: true });
    const {
      cardNumber,
      cardPassword,
      tipo,
      cardName,
      cardColor
    } = this.state;

    if (
      isNumeric(cardNumber) &&
      isNumeric(cardPassword) &&
      tipo === 'santander'
    ) {
      axios
        .get(
          `https://www.particulares.santandertotta.pt/bepp/sanpt/usuarios/loginrefeicao/?accion=3&identificacionUsuario=${cardNumber}&claveConsultiva=${cardPassword}`
        )
        .then(async ({ data }) => {
          const root = HTMLParser.parse(data);
          if (root.childNodes.length === 1) {
            try {
              AsyncStorage.setItem(
                cardNumber,
                JSON.stringify({
                  cardNumber,
                  cardPassword,
                  tipo,
                  cardName,
                  cardColor
                })
              ).then(() => this.props.navigation.navigate('Loading'));
            } catch (error) {
              this.setState({ message: error, isLoading: false });
            }
          } else {
            this.setState({
              message: 'Dados do cartão invalidos.',
              isLoading: false
            });
          }
        });
    }
    if (
      isNumeric(cardNumber) &&
      isNumeric(cardPassword) &&
      tipo === 'edenred'
    ) {
      const results = await loginEdenred(cardNumber, cardPassword);
      const root = HTMLParser.parse(results);

      const loginFailed = root.querySelector('span.rf-msgs-sum');

      if (loginFailed) {
        this.setState({ message: 'Dados do cartão invalidos.' });
      } else {
        try {
          AsyncStorage.setItem(
            cardNumber,
            JSON.stringify({
              cardNumber,
              cardPassword,
              tipo,
              cardName,
              cardColor
            })
          ).then(() => this.props.navigation.navigate('Loading'));
        } catch (error) {
          this.setState({ message: error });
        }
      }
    }

    this.setState({ isLoading: false });
    return null;
  }

  render() {
    return (
      <View style={{ flex: 1, paddingTop: 15 }}>
        <ScrollView>
          <View style={{ flex: 1, alignItems: 'center' }}>
            <CreditCard
              number={this.state.cardNumber}
              cvc={this.state.cardPassword}
              focused={this.state.isFocused}
              bgColor={this.state.cardColor}
              nome={this.state.cardName}
            />
            <Picker
              selectedValue={this.state.tipo}
              style={{ height: 50, width: 300, color: '#86899E' }}
              onValueChange={itemValue => this.setState({ tipo: itemValue })}
              itemStyle={{ fontWeight: 'bold', fontSize: 300 }}
            >
              <Picker.Item label="Santander" value="santander" />
              <Picker.Item label="Euroticket - Edenred" value="edenred" />
            </Picker>
            <View style={styles.circles}>
              <TouchableOpacity
                style={[styles.circle, { backgroundColor: '#F57A7A' }]}
                onPress={() => this.setState({ cardColor: '#F57A7A' })}
              />
              <TouchableOpacity
                style={[styles.circle, { backgroundColor: '#5CB15A' }]}
                onPress={() => this.setState({ cardColor: '#5CB15A' })}
              />
              <TouchableOpacity
                style={[styles.circle, { backgroundColor: '#2D5B99' }]}
                onPress={() => this.setState({ cardColor: '#2D5B99' })}
              />
              <TouchableOpacity
                style={[styles.circle, { backgroundColor: '#F0C41B' }]}
                onPress={() => this.setState({ cardColor: '#F0C41B' })}
              />
              <TouchableOpacity
                style={[styles.circle, { backgroundColor: '#A3A3A3' }]}
                onPress={() => this.setState({ cardColor: '#A3A3A3' })}
              />
            </View>
            <Input
              placeholder="XXXX XXXX XXXX XXXX"
              onChangeText={cardNumber => this.setState({ cardNumber })}
              value={this.state.cardNumber}
              maxLength={16}
              keyboardType="numeric"
              onFocus={() => this.setState({ isFocused: false, message: null })}
              inputContainerStyle={styles.input}
              containerStyle={{ marginTop: 15 }}
              label="Numero cartão"
            />
            <Input
              placeholder="XXX XXX"
              onChangeText={cardPassword => this.setState({ cardPassword })}
              value={this.state.cardPassword}
              maxLength={6}
              keyboardType="numeric"
              onFocus={() => this.setState({ isFocused: true, message: null })}
              inputContainerStyle={styles.input}
              containerStyle={{ marginTop: 15 }}
              label="CSV"
              rightIcon={
                <Icon
                  name="info-circle"
                  size={24}
                  color="grey"
                  onPress={() =>
                    this.setState({ isVisible: !this.state.isVisible })
                  }
                  underlayColor="white"
                />
              }
            />
            <Input
              placeholder="Nome opcional para o seu cartão"
              onChangeText={cardName => this.setState({ cardName })}
              value={this.state.cardName}
              maxLength={16}
              onFocus={() => this.setState({ isFocused: false })}
              inputContainerStyle={styles.input}
              containerStyle={{ marginTop: 15 }}
              label="Nome do cartão"
            />
            {this.state.message ? (
              <Text style={styles.warning}>{this.state.message}</Text>
            ) : null}
            <Button
              title="ADICIONAR"
              titleStyle={{ fontWeight: '700', color: 'grey' }}
              buttonStyle={{
                backgroundColor: 'rgba(255, 255, 255, 1)',
                width: 300,
                height: 45
              }}
              containerStyle={styles.button}
              loadingProps={{ color: this.state.cardColor }}
              loading={this.state.isLoading}
              onPress={() =>
                this.onPressValidator(
                  this.state.cardNumber,
                  this.state.cardPassword,
                  this.state.tipo
                )
              }
            />
          </View>
          <Overlay
            isVisible={this.state.isVisible}
            windowBackgroundColor="rgba(255, 255, 255, .8)"
            width="auto"
            height="auto"
            overlayStyle={styles.overlay}
            fullscreen
          >
            <Image source={require('../assets/images/example.png')} />
            <Icon
              name="times"
              size={30}
              color="black"
              onPress={() =>
                this.setState({ isVisible: !this.state.isVisible })
              }
              containerStyle={styles.close}
              style={styles.close}
              raised
            />
          </Overlay>
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  input: {
    borderColor: 'grey',
    borderBottomWidth: 1,
    marginBottom: 5
  },
  overlay: {
    position: 'absolute',
    right: 100,
    top: 0
  },
  close: {
    position: 'absolute',
    top: -2,
    right: 0
  },
  button: {
    marginTop: 50
  },
  warning: {
    marginTop: 15,
    color: 'rgba(255, 0, 0, .8)',
    fontWeight: 'bold'
  },
  circles: {
    display: 'flex',
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-around',
    marginTop: 10,
    marginBottom: 10
  },
  circle: {
    height: 30,
    width: 30,
    borderRadius: 50
  }
});

AddCard.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired
};
