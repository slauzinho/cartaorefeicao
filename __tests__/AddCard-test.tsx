import React from 'react';
import {
  render,
  fireEvent,
  flushMicrotasksQueue,
  waitForElement,
} from 'react-native-testing-library';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import AddCard from '../components/AddCard';
import reducer from '../reducers';
import { loginEdenred, loginSantander } from '../utilities/login';
import { mocked } from 'ts-jest/utils';
import { Button } from 'react-native-elements';
import { addCardRequest } from '../actions';
import { Picker } from 'react-native';
import { act } from 'react-test-renderer';
import CreditCard from '../components/CreditCard';

jest.mock('../utilities/login');
jest.mock('../actions');

const mockedLoginEdenred = mocked(loginEdenred, true);
const mockedLoginSantander = mocked(loginSantander, true);

const createTestProps = (props?: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) }: any = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

describe('AddCard', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it('renders the screen correctly', () => {
    const props = createTestProps();
    const { toJSON } = renderWithRedux(<AddCard {...(props as any)} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('allows me to submit my santander card details', async () => {
    mockedLoginSantander.mockResolvedValueOnce({
      saldo: '100 €',
      transactions: [],
    });
    const props = createTestProps();
    const redux = require('react-redux');
    redux.useDispatch = jest.fn(() => jest.fn());

    const { getByType, getByText } = renderWithRedux(
      <AddCard {...(props as any)} />
    );

    fireEvent.changeText(getByText(/Numero cartão/i), '2424242424242424');
    fireEvent.changeText(getByText(/CSV/i), '24242424');
    fireEvent.press(getByType(Button));

    await flushMicrotasksQueue();

    expect(props.navigation.navigate).toHaveBeenCalledTimes(1);
    expect(props.navigation.navigate).toHaveBeenCalledWith('Loading');
    expect(addCardRequest).toHaveBeenCalledWith({
      cardColor: '#F57A7A',
      cardName: '',
      cardNumber: '2424242424242424',
      cardPassword: '24242424',
      email: '',
      tipo: 'santander',
    });
    expect(mockedLoginSantander).toHaveBeenCalledTimes(1);
  });

  it('allows me to submit my edenred card details', async () => {
    mockedLoginEdenred.mockResolvedValueOnce({
      saldo: '100 €',
      transactions: [],
    });
    const props = createTestProps();
    const redux = require('react-redux');
    redux.useDispatch = jest.fn(() => jest.fn());

    const { getByType, getByText } = renderWithRedux(
      <AddCard {...(props as any)} />
    );

    fireEvent(getByType(Picker), 'onValueChange', 'edenred');

    fireEvent.changeText(getByText(/Numero cartão/i), '2424242424242424');
    fireEvent.changeText(getByText(/Email/i), 'test@example.com');
    fireEvent.changeText(getByText(/Password/i), 'randomPassword');
    fireEvent.press(getByType(Button));

    await flushMicrotasksQueue();

    expect(props.navigation.navigate).toHaveBeenCalledTimes(1);
    expect(props.navigation.navigate).toHaveBeenCalledWith('Loading');
    expect(addCardRequest).toHaveBeenCalledWith({
      cardColor: '#F57A7A',
      cardName: '',
      cardNumber: '2424242424242424',
      cardPassword: 'randomPassword',
      email: 'test@example.com',
      tipo: 'edenred',
    });
    expect(mockedLoginSantander).toHaveBeenCalledTimes(0);
    expect(mockedLoginEdenred).toHaveBeenCalledTimes(1);
  });

  it('handles wrong login details for santander', async () => {
    mockedLoginSantander.mockRejectedValueOnce('Fail to login');
    const props = createTestProps();
    const redux = require('react-redux');
    redux.useDispatch = jest.fn(() => jest.fn());

    const { getByType, getByText } = renderWithRedux(
      <AddCard {...(props as any)} />
    );

    fireEvent.changeText(getByText(/Numero cartão/i), '2424242424242424');
    fireEvent.changeText(getByText(/CSV/i), '24242424');
    fireEvent.press(getByType(Button));

    await waitForElement(() => getByText('Falhou ao tentar fazer login'));
  });

  it('handles wrong login details for edenred', async () => {
    mockedLoginEdenred.mockRejectedValueOnce('Fail to login');
    const props = createTestProps();
    const redux = require('react-redux');
    redux.useDispatch = jest.fn(() => jest.fn());

    const { getByType, getByText } = renderWithRedux(
      <AddCard {...(props as any)} />
    );

    fireEvent(getByType(Picker), 'onValueChange', 'edenred');

    fireEvent.changeText(getByText(/Numero cartão/i), '2424242424242424');
    fireEvent.changeText(getByText(/Email/i), 'test@example.com');
    fireEvent.changeText(getByText(/Password/i), 'randomPassword');
    fireEvent.press(getByType(Button));

    await waitForElement(() => getByText('Falhou ao tentar fazer login'));
  });

  it('allows me to choose different colours', async () => {
    const props = createTestProps();
    const redux = require('react-redux');
    redux.useDispatch = jest.fn(() => jest.fn());

    const { getByTestId, getByType } = renderWithRedux(
      <AddCard {...(props as any)} />
    );

    fireEvent.press(getByTestId('pinkColorBtn'));
    expect(getByType(CreditCard).props.bgColor).toBe('#F57A7A');
    fireEvent.press(getByTestId('greenColorBtn'));
    expect(getByType(CreditCard).props.bgColor).toBe('#5CB15A');
    fireEvent.press(getByTestId('blueColorBtn'));
    expect(getByType(CreditCard).props.bgColor).toBe('#2D5B99');
    fireEvent.press(getByTestId('yellowColorBtn'));
    expect(getByType(CreditCard).props.bgColor).toBe('#F0C41B');
    fireEvent.press(getByTestId('greyColorBtn'));
    expect(getByType(CreditCard).props.bgColor).toBe('#A3A3A3');
  });
});
