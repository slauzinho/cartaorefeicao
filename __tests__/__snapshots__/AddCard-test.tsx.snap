// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard renders the screen correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 15,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
        >
          <View
            alignHeight={false}
            alignWidth={false}
            clickable={true}
            flip={false}
            flipHorizontal={true}
            flipVertical={false}
            friction={6}
            onFlipEnd={[Function]}
            onFlipStart={[Function]}
            perspective={1000}
            style={
              Object {
                "backgroundColor": "#F57A7A",
                "borderRadius": 8,
                "borderWidth": 0,
                "flex": 0,
                "height": 190,
                "opacity": 1,
                "transform": Array [
                  Object {
                    "perspective": 1000,
                  },
                  Object {
                    "rotateY": "0deg",
                  },
                ],
                "width": 300,
              }
            }
            useNativeDriver={true}
          >
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Array [
                    false,
                    false,
                  ],
                ]
              }
            >
              <View
                style={
                  Object {
                    "height": 190,
                    "width": 300,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Array [
                      Object {
                        "borderRadius": 8,
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "height": 190,
                        "width": 300,
                      },
                    ]
                  }
                />
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 2,
                        "justifyContent": "center",
                        "paddingTop": 40,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#fff",
                          "fontSize": 21,
                          "marginBottom": 10,
                          "textAlign": "center",
                          "textShadowColor": "rgba(0, 0, 0, 0.75)",
                          "textShadowOffset": Object {
                            "height": 1,
                            "width": 1,
                          },
                          "textShadowRadius": 1,
                        }
                      }
                    >
                      •••• •••• •••• ••••
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "bottom": 20,
                      "left": 32,
                      "position": "absolute",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontSize": 21,
                        "marginBottom": 10,
                        "textAlign": "center",
                        "textShadowColor": "rgba(0, 0, 0, 0.75)",
                        "textShadowOffset": Object {
                          "height": 1,
                          "width": 1,
                        },
                        "textShadowRadius": 1,
                      }
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 50,
              "width": 300,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Santander",
                  "textColor": undefined,
                  "value": "santander",
                },
                Object {
                  "label": "Euroticket - Edenred",
                  "textColor": undefined,
                  "value": "edenred",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                Object {
                  "fontSize": 300,
                  "fontWeight": "bold",
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-around",
              "marginBottom": 10,
              "marginTop": 10,
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#F57A7A",
                "borderRadius": 50,
                "height": 30,
                "opacity": 1,
                "width": 30,
              }
            }
            testID="pinkColorBtn"
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#5CB15A",
                "borderRadius": 50,
                "height": 30,
                "opacity": 1,
                "width": 30,
              }
            }
            testID="greenColorBtn"
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#2D5B99",
                "borderRadius": 50,
                "height": 30,
                "opacity": 1,
                "width": 30,
              }
            }
            testID="blueColorBtn"
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#F0C41B",
                "borderRadius": 50,
                "height": 30,
                "opacity": 1,
                "width": 30,
              }
            }
            testID="yellowColorBtn"
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#A3A3A3",
                "borderRadius": 50,
                "height": 30,
                "opacity": 1,
                "width": 30,
              }
            }
            testID="greyColorBtn"
          />
        </View>
        <View />
        <View />
        <View
          style={
            Object {
              "marginTop": 15,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#86939e",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Numero cartão
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "grey",
                "flexDirection": "row",
                "marginBottom": 5,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              keyboardType="numeric"
              maxLength={16}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="XXXX XXXX XXXX XXXX"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 15,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#86939e",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            CSV
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "grey",
                "flexDirection": "row",
                "marginBottom": 5,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              keyboardType="numeric"
              maxLength={6}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="XXX XXX"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 15,
                }
              }
            >
              <Text
                allowFontScaling={false}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "color": "grey",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="openModalBtn"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 15,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#86939e",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Nome do cartão
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderBottomWidth": 1,
                "borderColor": "grey",
                "flexDirection": "row",
                "marginBottom": 5,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              editable={true}
              maxLength={16}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Nome opcional para o seu cartão"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "black",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
              "marginTop": 50,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "marginBottom": 60,
                  "padding": 8,
                  "width": 300,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 18,
                    "fontWeight": "700",
                    "paddingBottom": 1,
                    "paddingTop": 2,
                    "textAlign": "center",
                  }
                }
              >
                ADICIONAR
              </Text>
            </View>
          </View>
        </View>
      </View>
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        hideModalContentWhileAnimating={false}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
        transparent={true}
        visible={false}
      >
        <View
          accessible={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "height": 1334,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 750,
            }
          }
        />
        <View
          hideModalContentWhileAnimating={false}
          onModalHide={[Function]}
          onModalWillHide={[Function]}
          onModalWillShow={[Function]}
          pointerEvents="box-none"
          scrollHorizontal={false}
          scrollOffset={0}
          scrollOffsetMax={0}
          scrollTo={null}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
          supportedOrientations={
            Array [
              "portrait",
              "landscape",
            ]
          }
          swipeThreshold={100}
        >
          <Image
            resizeMode="contain"
            source={1}
          />
          <Text />
        </View>
      </Modal>
    </View>
  </RCTScrollView>
</View>
`;
